OpenSim/Common/Component.h

```diff
@@ -93,6 +93,39 @@ class ComponentNotFoundOnSpecifiedPath : public Exception {
     }
 };
 
+class ComponentIsAnOrphan : public Exception {
+public:
+    ComponentIsAnOrphan(const std::string& file,
+        size_t line,
+        const std::string& func,
+        const std::string& thisName,
+        const std::string& componentConcreteClassName) :
+        Exception(file, line, func) {
+        std::string msg = "Component '" + thisName + "' of type " +
+            componentConcreteClassName + " has no parent and is not the root.\n" +
+            "Verify that finalizeFromProperties() has been invoked on the " + 
+            "root Component or that this Component is not a clone, which has " +
+            "not been added to its parent.";
```

Comment:
Consider clarifying "parent" as "parent component" to avoid confusion with "parent frames" etc.